test/e2e/test_helpers.go

```diff
@@ -232,6 +236,60 @@ func (t testHelper) createUUIDProjectNMT(ctx context.Context, clusterName, names
 	return nmt
 }
 
+type createGPUNMTParams struct {
+	gpuVendorEnvKey string
+	gpuNameEnvKey   string
+}
+
+func (t testHelper) createNameGPUNMT(ctx context.Context, clusterName, namespace string, params createGPUNMTParams) *infrav1.NutanixMachineTemplate {
+	gpuName := t.getVariableFromE2eConfig(params.gpuNameEnvKey)
+
+	nmt := t.createDefaultNMT(clusterName, namespace)
+	nmt.Spec.Template.Spec.GPUs = []infrav1.NutanixGPU{
+		{
+			Type: infrav1.NutanixGPUIdentifierName,
+			Name: &gpuName,
+		},
+	}
+	return nmt
+}
+
+func (t testHelper) findGPU(ctx context.Context, gpuName string) *prismGoClientV3.GPU {
+	clusterVarValue := t.getVariableFromE2eConfig(clusterVarKey)
+
+	clusterUUID, err := controllers.GetPEUUID(ctx, t.nutanixClient, &clusterVarValue, nil)
+	Expect(err).ToNot(HaveOccurred())
+	Expect(clusterUUID).ToNot(BeNil())
+	allGpus, err := controllers.GetGPUsForPE(ctx, t.nutanixClient, clusterUUID)
+	Expect(err).ToNot(HaveOccurred())
+	Expect(allGpus).ToNot(HaveLen(0))
```

Comment:
IMO, this assertion should fail with a message stating this test relies on GPU being associated with PE and is expected to fail on PE without GPU.